modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/upgrade/v3_1_0/UrlSubjectUpgradeProcess.java

```diff
@@ -80,26 +86,28 @@ private void _populateUrlSubject() throws Exception {
 					"update MBMessage set urlSubject = ? where messageId = " +
 						"?")) {
 
-			int count = 0;
 			String curURLSubject = null;
-			String previousURLSubject = null;
 
 			while (resultSet.next()) {
 				long messageId = resultSet.getLong(1);
 				String subject = resultSet.getString(2);
 
+				String suffix = StringPool.BLANK;
+
 				curURLSubject = _getURLSubject(messageId, subject);
 
-				String suffix = null;
+				IntegerWrapper count = counts.getOrDefault(
+					curURLSubject, new IntegerWrapper(0));
+
+				if (count.getValue() > 0) {
+					suffix = StringPool.DASH + count.getValue();
+
+					counts.put(curURLSubject + suffix, new IntegerWrapper(1));
 
-				if (StringUtil.equals(previousURLSubject, curURLSubject)) {
-					count++;
-					suffix = StringPool.DASH + count;
+					count.increment();
 				}
 				else {
-					count = 0;
-					previousURLSubject = curURLSubject;
-					suffix = StringPool.BLANK;
+					counts.put(curURLSubject, new IntegerWrapper(1));
```

Comment:
Here, I'd do something like this to avoid creating new IntegerWrapper instances every time you do the `getOrDefault`:
```
IntegerWrapper count = counts.computeIfAbsent(..., () -> new IntegerWrapper(0));

if (count.getValue() > 0) {
    ...
}
else {
    count.increment();  // instance already in the map, so no need to add a new one again.
}
```

Reply:
thanks!!